[quant][graphmode] Test JIT tracing for dynamic quant cases

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40040 [quant][graphmode] Test JIT tracing for dynamic quant cases**
* #40039 [quant][graphmode] Refactor dynamic quant tests

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22071277](https://our.internmc.facebook.com/intern/diff/D22071277)

supriyar committed:
[quant][graphmode] Test JIT tracing for dynamic quant cases

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2461,14 +2461,19 @@ def __init__(self):
-                   .run(model.graph)
+        eager_model = M().eval()
+        for tracing_mode in [True, False]:
+            if tracing_mode:
+                x = torch.randn(5, 5)
+                model = torch.jit.trace(eager_model, x)
+            else:
+                model = torch.jit.script(eager_model)
```
we have a utility function for this: `get_script_module`. please take a look at other tests in the file.

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2461,14 +2461,19 @@ def __init__(self):
-                   .check("quantized::linear_dynamic") \
-                   .run(model.graph)
+        eager_model = M().eval()
+        for tracing_mode in [True, False]:
```
nit: suggest to use `tracing` for the name to align with other tests in the file

supriyar committed:
Update on "[quant][graphmode] Test JIT tracing for dynamic quant cases"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2461,14 +2461,16 @@ def __init__(self):
+        x = torch.randn(5, 5)
+        for tracing in [True, False]:
+            model = get_script_module(eager_model, tracing, x)
+            qconfig_dict = {'' : default_dynamic_qconfig}
+            model = quantize_dynamic_script(model, qconfig_dict)
+            # add op is not dynamically quantized.
+            FileCheck().check("aten::add") \
```
can you use `_test_op_impl`? It is in TestQuantizeScriptPTSQOps, but you can move it to QuantizationTestCase

supriyar replied:
Sure. That currently only tests `quantize_script` API. I'll make it generic.

jerryzh168 replied:
or maybe define a `_test_op_impl` locally for this test

supriyar replied:
@jerryzh168 refactored it a bit to make it usable by both.

supriyar committed:
Update on "[quant][graphmode] Test JIT tracing for dynamic quant cases"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2462,13 +2430,13 @@ def forward(self, x):
-        FileCheck().check("aten::add") \
-                   .check("quantized::linear_dynamic") \
-                   .run(model.graph)
+        eager_model = M().eval()
```
btw, `eval_mode` is an argument of `checkGraphModeOp`

jerryzh168 reviewed (approved):
LG, thanks!

supriyar committed:
Update on "[quant][graphmode] Test JIT tracing for dynamic quant cases"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Test JIT tracing for dynamic quant cases"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22071277](https://our.internmc.facebook.com/intern/diff/D22071277)

[ghstack-poisoned]
